frame/staking/src/pallet/mod.rs

```diff
@@ -324,12 +324,27 @@ pub mod pallet {
 
 	/// Where the reward payment should be made. Keyed by stash.
 	///
+	/// NOTE: Being lazily migrated and deprecated in favour of `Payees`.
+	/// Tracking at <https://github.com/paritytech/substrate/issues/14438>
 	/// TWOX-NOTE: SAFE since `AccountId` is a secure hash.
 	#[pallet::storage]
 	#[pallet::getter(fn payee)]
 	pub type Payee<T: Config> =
 		StorageMap<_, Twox64Concat, T::AccountId, RewardDestination<T::AccountId>, ValueQuery>;
 
+	/// Where the reward payment should be made. Keyed by stash.
+	///
+	/// TWOX-NOTE: SAFE since `AccountId` is a secure hash.
+	#[pallet::storage]
+	#[pallet::getter(fn payees)]
+	pub type Payees<T: Config> = CountedStorageMap<
```

Comment:
Why do we need a *counted* storage map?

Reply:
The discussion we had about verifying the number of `Payees` in try_state or future migrations is something that would benefit from using a counted storage map. Beyond this, a map with 68k records that will grow further as the staking system scales would benefit from a counter, as to not pull the entire storage item for analytics tools or if on-chain logic needs it.